Support PAPI versions greater 5.4.1

Currently it is not possible to compile DASH with a PAPI version > 5.4.1. The problem cannot be easily fixed, as functions from `string.h` and `papi.h` clash.

```
In file included from /usr/include/c++/5/cstring:42:0,
                 from /opt/dash/dash/include/dash/internal/StreamConversion.h:14,
                 from /opt/dash/dash/include/dash/internal/Logging.h:5,
                 from /opt/dash/dash/include/dash/util/Timer.h:20,
                 from /opt/dash/dash/src/util/Timer.cc:1:
/usr/include/string.h:525:40: error: declaration of 'int ffsll(long long int) throw ()' has a different exception specifier
      __THROW __attribute__ ((__const__));
                                        ^
In file included from /opt/dash/dash/include/dash/util/internal/TimestampPAPI.h:9:0,
                 from /opt/dash/dash/include/dash/util/Timer.h:14,
                 from /opt/dash/dash/src/util/Timer.cc:1:
/opt/papi/include/papi.h:1024:8: error: from previous declaration 'int ffsll(long long int)'
    int ffsll(long long lli); //required for --with-ffsll and used in extras.c/papi.c
```

--- a/dash/scripts/docker-testing/mpich/dockerfile
+++ b/dash/scripts/docker-testing/mpich/dockerfile
@@ -13,17 +13,14 @@ RUN           apt-get install -y \
                   libmpich-dev \
                   libhdf5-mpich-dev
 
-# Install PAPI 5.5.0 from source
+# Install PAPI from source
 WORKDIR       /tmp
 
-# Latest version of PAPI is not compatible
-# ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.5.0.tar.gz papi.tgz
-
-ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.4.1.tar.gz papi.tgz
+ADD           http://icl.utk.edu/projects/papi/downloads/papi-5.5.1.tar.gz papi.tgz
 RUN           tar -xf papi.tgz
-RUN           cd papi*/src/                                  \
-              && ./configure --prefix=/opt/papi --with-ffsll \
-              && make                                        \
+RUN           cd papi*/src/                     \
+              && ./configure --prefix=/opt/papi \
+              && make                           \
               && make install
 ENV           PAPI_BASE=/opt/papi
 ENV           LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${PAPI_BASE}/lib

--- a/dash/scripts/docker-testing/openmpi/dockerfile
+++ b/dash/scripts/docker-testing/openmpi/dockerfile
@@ -15,17 +15,14 @@ RUN           apt-get install -y \
                   openmpi-common \
                   libhdf5-openmpi-dev
 
-# Install PAPI 5.5.0 from source
+# Install PAPI from source
 WORKDIR       /tmp
 
-# Latest version of PAPI is not compatible
-# ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.5.0.tar.gz papi.tgz
-
-ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.4.1.tar.gz papi.tgz
+ADD           http://icl.utk.edu/projects/papi/downloads/papi-5.5.1.tar.gz papi.tgz
 RUN           tar -xf papi.tgz
-RUN           cd papi*/src/                                  \
-              && ./configure --prefix=/opt/papi --with-ffsll \
-              && make                                        \
+RUN           cd papi*/src/                     \
+              && ./configure --prefix=/opt/papi \
+              && make                           \
               && make install
 ENV           PAPI_BASE=/opt/papi
 ENV           LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${PAPI_BASE}/lib

--- a/dash/scripts/docker-testing/openmpi2/dockerfile
+++ b/dash/scripts/docker-testing/openmpi2/dockerfile
@@ -11,17 +11,14 @@ RUN           apt-get install -y \
                   libhwloc-plugins \
                   libhwloc-dev
 
-# Install PAPI 5.5.0 from source
+# Install PAPI from source
 WORKDIR       /tmp
 
-# Latest version of PAPI is not compatible
-# ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.5.0.tar.gz papi.tgz
-
-ADD           http://icl.cs.utk.edu/projects/papi/downloads/papi-5.4.1.tar.gz papi.tgz
+ADD           http://icl.utk.edu/projects/papi/downloads/papi-5.5.1.tar.gz papi.tgz
 RUN           tar -xf papi.tgz
-RUN           cd papi*/src/                                  \
-              && ./configure --prefix=/opt/papi --with-ffsll \
-              && make                                        \
+RUN           cd papi*/src/                     \
+              && ./configure --prefix=/opt/papi \
+              && make                           \
               && make install
 # Openmpi 2.0
 ADD           https://www.open-mpi.org/software/ompi/v2.0/downloads/openmpi-2.0.1.tar.gz ompi.tgz

